docs/data/data-grid/row-recipes/DetailPanelCollapseAll.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import UnfoldLessIcon from '@mui/icons-material/UnfoldLess';
+import {
+  DataGridPro,
+  GridColDef,
+  GridRowsProp,
+  GridRowParams,
+  useGridApiContext,
+  useGridSelector,
+  gridDetailPanelExpandedRowIdsSelector,
+  GRID_DETAIL_PANEL_TOGGLE_COL_DEF,
+} from '@mui/x-data-grid-pro';
+import {
+  randomCreatedDate,
+  randomCurrency,
+  randomEmail,
+  randomPrice,
+} from '@mui/x-data-grid-generator';
+
+export default function DetailPanelCollapseAll() {
+  const getDetailPanelContent = React.useCallback(
+    ({ row }: GridRowParams) => <Box sx={{ p: 2 }}>{`Order #${row.id}`}</Box>,
+    [],
+  );
+
+  const getDetailPanelHeight = React.useCallback(() => 50, []);
+
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGridPro
+        rows={rows}
+        columns={columns}
+        getDetailPanelContent={getDetailPanelContent}
+        getDetailPanelHeight={getDetailPanelHeight}
+      />
+    </div>
+  );
+}
+
+function CustomDetailPanelHeader() {
+  const apiRef = useGridApiContext();
+  const expandedRowIds = useGridSelector(
+    apiRef,
+    gridDetailPanelExpandedRowIdsSelector,
+  );
+  const areDetailPanelsOpen = expandedRowIds.length > 0;
+
+  const closeAllDetailPanels = () => apiRef.current.setExpandedDetailPanels([]);
```

Comment:
How about covering "expand all" here as well?

Reply:
Done in c33ff435a9d868b5ae29a490c52508a70e70a045